tutorials/shaders/shader_reference/shader_preprocessor.rst

```diff
@@ -0,0 +1,141 @@
+.. _doc_shader_preprocessor:
+
+Shader Preprocessor
+===================
+
+Shader preprocessor belongs as optional step before shader compilation of text shaders in Godot.
```

Comment:
IMO we should start this page with some examples of use cases / what this is useful for, especially for people that are new to this.

Reply:
I will think about it.